### PR TITLE
support customized payout period

### PR DESCRIPTION
**Pull request by laipogo**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior?** (You can also refer to a JIRA ticket here)

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **What dependencies / modules need to be updated as pre-requisites?** (Include relevant PR links here)

* **Description of changes**

---

* **Please check if the PR fulfills these requirements**
- [ ] The subject of this PR contains a JIRA ticket BTFS-xxxx (N/A for community PRs)
- [ ] PR description and commit messages are [good and meaningful](https://chris.beams.io/posts/git-commit/)
- [ ] Unit tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **Please make sure the following procedures have been applied before requesting reviewers**
- [ ] Code changes closely follow [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
- [ ] Code has been *rebased* against latest master (`git merge` not recommended, unless you know what you are doing)
- [ ] Code changes have run through `go fmt`
- [ ] Code changes have run through `go mod tidy`
- [ ] All unit tests passed locally (`make test_go_test`)

### CODE REVIEW

File: core/commands/storage.go [hunk 1]
```diff
@@ -52,9 +53,13 @@ const (
...
 	repairModeOptionName          = "repair-mode"
+	payoutScheduleOptionName      = "payout-schedule"
+	numPayoutsOptionName          = "num-payouts"
 
-	defaultRepFactor     = 3
-	defaultStorageLength = 30
+	defaultRepFactor      = 3
+	defaultStorageLength  = 30
+	defaultPayoutSchedule = 0
+	defaultNumPayouts     = 1
```
Comment (taiyangc) [inline core/commands/storage.go]: 1 payout by default??

File: core/escrow/escrow.go [hunk 2]
```diff
@@ -52,8 +52,8 @@ func NewContract(configuration *config.Config, id string, n *core.IpfsNode, pid
 		CollateralAmount: 0,
 		WithholdAmount:   0,
 		TokenType:        escrowpb.TokenType_BTT,
-		PayoutSchedule:   0,
-		NumPayouts:       1,
+		PayoutSchedule:   payoutSchedule,
```
Comment (jin-tron) [inline core/escrow/escrow.go]: don't see where you use the   int32 customize_payout_period = 12; // customized payout period in days
if this is not set, we will still use current 30 days as pay out period.
enum Schedule {
  MONTHLY = 0;
  QUARTERLY = 1;
  ANNUALLY = 2;
  CUSTOMIZED = 3;
}

File: core/commands/storage.go (hunk 1 above)
Comment (taiyangc) [reply core/commands/storage.go]: as jin mentioned, you need to use constants, and also update go-btfs-common to latest release

File: core/commands/storage.go [hunk 3]
```diff
@@ -455,7 +472,7 @@ func retryMonitor(ctx context.Context, api coreiface.CoreAPI, ss *storage.FileCo
 			}
 
 			// build connection with host, init step, could be error or timeout
-			go retryProcess(param, nil, true)
+			go retryProcess(param, nil, true, n, customizedSchedule, period)
```
Comment (taiyangc) [inline core/commands/storage.go]: put the two new params inside `param` so we don't have to modify it further in the future

File: core/commands/storage.go [hunk 4]
```diff
@@ -333,7 +338,17 @@ Use status command to check for completion:
 		if req.Options[testOnlyOptionName] != nil {
 			testFlag = req.Options[testOnlyOptionName].(bool)
 		}
-		go retryMonitor(context.Background(), api, ss, n, ssID, testFlag, isRepairMode, renterPid.Pretty())
+		p := 1
```
Comment (taiyangc) [inline core/commands/storage.go]: can you put this number on line 139 with a `IntOption.(...).WithDefault(defaultPayoutPeriod)`?, then you don't need line 344-347

**Pull request merged**